### PR TITLE
auth functionality, redux ,reduxthunk, validations

### PR DESCRIPTION
**Pull request by kgajriaAgarwal**
Hello, i have Added auth functionality, redux ,reduxthunk, validations in social media app. kindly review it and let me knoe what do think about it !
files to check:
src/Helpers/Validations/ValidationMessages.js
src/app/slice/authSlice.js
src/app/store/store.js
src/pages/Auth/Login.js
src/pages/Auth/SignUp.js
PR: https://github.com/kgajriaAgarwal/blithe-in/pull/3
PREVIEW - https://blithe-in.netlify.app/

### CODE REVIEW

**Review by Naman-Saxena1 — APPROVED** [review]
Code looks fine and functionality is also working fine in preview.
My only suggestion would be, I understand you might need some commented code in future PRs, so you can make sure that there is no commented code in the last PR. Because currently there is a lot of commented code in it 😅

File: src/App.js [hunk 1]
```diff
@@ -61,6 +61,11 @@ function App() {
       <Box className="App" bgcolor={"Background.default"}>
         <BlitheRoute/>
       </Box>
+      <ToastContainer
+      autoClose={4000}
```
Comment (mikesch803) [inline src/App.js]: suggestion : toast taking too much time, you should change this

File: src/Helpers/Services/apiJson.jsx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+const encodedToken = localStorage.getItem("authData") ? localStorage.getItem("authData") : '';
+
+export const ApiJson = { 
+
+//ACTION TO LOGIN
+actionLogin: {
+  url: '/auth/login',
+  method: 'POST',
+  data: {
+    // "email": "",
```
Comment (mikesch803) [inline src/Helpers/Services/apiJson.jsx]: you can remove if you are not using email

File: src/Helpers/Services/apiJson.jsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+      data: {},
+      headers: {
+        'Content-Type': 'application/json',
+        'authorization': encodedToken
+      }, 
+      showResultMessage: false,
+      showErrorMessage: true,
+    },
+
+    //action to clear all history
```
Comment (mikesch803) [inline src/Helpers/Services/apiJson.jsx]: Remove : you have added video library services to this app

**Review by mikesch803 — APPROVED** [review]
since you have added authentication, you should wrap home in private routes, and put some suggestion, please look into it, apart from it, nice ui, dark mode is nice.

File: src/Helpers/Services/apiJson.jsx (hunk 2 above)
Comment (kgajriaAgarwal) [reply src/Helpers/Services/apiJson.jsx]: removed

File: src/App.js (hunk 1 above)
Comment (kgajriaAgarwal) [reply src/App.js]: changed!

File: src/Helpers/Common/utils.js [hunk 4]
```diff
@@ -1,10 +1,61 @@
...
+    // const encodedData = encryptedData(value);
+    localStorage.setItem(key, value);
+  };
+  
+  // used to get localstorage item
+  export const getLocalStorage = (key) => {
+    if (key) {
+      let data = localStorage.getItem(key);
+      if (data) {
+        // data = JSON.parse(decryptedData(data));
```
Comment (MayankKumar10) [inline src/Helpers/Common/utils.js:45]: Suggestion: remove this comment

File: src/Helpers/Common/utils.js [hunk 5]
```diff
@@ -1,10 +1,61 @@
...
+  
+  // toastr messages for success
+  export const showSuccessToast = (message) => { 
+    if (!toast.isActive(toastId)) {    
+      toastId = toast.success(message, {
+        position: toast.POSITION.TOP_RIGHT,
+        hideProgressBar: true,
+        closeOnClick: true,
+        pauseOnHover: true,
+        autoClose: 4000,
```
Comment (MayankKumar10) [inline src/Helpers/Common/utils.js]: Suggestion: Reduce the time in Toast

File: src/Helpers/Services/apiService.jsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
+//code is commented in this file for future perspective..
+////console.log is also commented , until the time the this app is fully developed
+import axios from 'axios';
+import { ApiJson } from './apiJson';
+
+// import { ApiJson } from './apiJson';
+
+// import {
```
Comment (MayankKumar10) [inline src/Helpers/Services/apiService.jsx:8]: Suggestion: remove this comment if it's not needed

**Review by MayankKumar10 — COMMENTED** [review]
Look into these suggestions everything else looks good Nice UI, you can merge it.

**Pull request merged**